src/ext/NetFx/ca/netfxca.cpp

```diff
@@ -879,3 +907,132 @@ extern "C" UINT __stdcall ExecNetFx(
+    LPWSTR pwzPlatform = NULL;
+    LPWSTR pwzNetCoreCheckBinaryId = NULL;
+    LPWSTR pwzNetCoreCheckFilePath = NULL;
+    LPWSTR pwzRuntimeType = NULL;
+    LPWSTR pwzVersion = NULL;
+    LPWSTR pwzRollForward = NULL;
+    LPWSTR pwzVariable = NULL;
+    LPWSTR pwzCommandLine = NULL;
+
+    // Initialize
+    hr = WcaInitialize(hInstall, "DotNetCompatibilityCheck");
+    ExitOnFailure(hr, "failed to initialize");
+
+    // Get temp directory path
+    dwTempPathLength = GetTempPathW(dwTempPathLength, NULL);
+    if (0 == dwTempPathLength)
+    {
+        hr = HRESULT_FROM_WIN32(GetLastError());
+        ExitOnFailure(hr, "failed to get temp path length");
+    }
+
+    hr = StrAlloc(&pwzTempPath, ++dwTempPathLength);
+    ExitOnFailure(hr, "failed to allocate temp path variable");
+
+    dwTempPathLength = GetTempPathW(dwTempPathLength, pwzTempPath);
+    if (0 == dwTempPathLength)
+    {
+        hr = HRESULT_FROM_WIN32(GetLastError());
+        ExitOnFailure(hr, "failed to get temp path value");
+    }
+
+    // Open view on .NET compatibility check table
+    hr = WcaOpenExecuteView(vscDotNetCompatibilityCheckQuery, &hView);
+    ExitOnFailure(hr, "failed to open view on Wix4NetFxDotNetCompatibilityCheck table");
+
+    // Go through all records and run NetCorCheck.exe for each
+    while (S_OK == (hr = WcaFetchRecord(hView, &hRec)))
+    {
+        HANDLE hProcess = INVALID_HANDLE_VALUE;
+        DWORD dwExitCode = 0;
+
+        // Extract NetCoreCheck.exe for platform to temp directory
+        hr = WcaGetRecordString(hRec, platform, &pwzPlatform);
+        ExitOnFailure(hr, "failed to get Wix4DotNetCompatibilityCheck.Platform");
+
+        hr = GetNetCoreCheckBinaryId(pwzPlatform, &pwzNetCoreCheckBinaryId);
+        ExitOnFailure(hr, "failed to get NetCoreCheck binary id for platform %ls", pwzPlatform);
+
+        hr = StrAllocFormatted(&pwzNetCoreCheckFilePath, L"%ls%ls", pwzTempPath, L"NetCoreCheck.exe");
+        ExitOnFailure(hr, "failed to set NetCoreCheck file path");
+
+        hr = WcaExtractBinaryToFile(pwzNetCoreCheckBinaryId, pwzNetCoreCheckFilePath);
+        ExitOnFailure(hr, "failed to extract NetCoreCheck from binary '%ls' to file %ls", pwzNetCoreCheckBinaryId, pwzNetCoreCheckFilePath);
+
+        // Read all NetCoreCheck.exe parameters and variable
+        hr = WcaGetRecordString(hRec, runtimeType, &pwzRuntimeType);
+        ExitOnFailure(hr, "failed to get Wix4DotNetCompatibilityCheck.RuntimeType");
+
+        hr = WcaGetRecordString(hRec, version, &pwzVersion);
+        ExitOnFailure(hr, "failed to get Wix4DotNetCompatibilityCheck.Version");
+
+        hr = WcaGetRecordString(hRec, rollForward, &pwzRollForward);
+        ExitOnFailure(hr, "failed to get Wix4DotNetCompatibilityCheck.RollForward");
+
+        hr = WcaGetRecordString(hRec, variable, &pwzVariable);
+        ExitOnFailure(hr, "failed to get Wix4DotNetCompatibilityCheck.Variable");
+
+        // Run NetCoreCheck.exe and store its result in variable
```

Comment:
@rseanhall just did something like this in Burn. Probably be good if he takes a look at it.

Reply:
Ok, but MSI custom actions aren't my strong suit. This appears to be doing what it's supposed to: run Microsoft's NetCoreCheck.exe and store the exit code in the property that the user requested.